sys/behave/src/Expand.hs

```diff
@@ -106,20 +117,24 @@ expand chsets (BNbexpr we (Parallel chans bexps))  =
 
 expand chsets (BNbexpr we (Enable bexp1 chanoffs bexp2))  =  do
      chanoffs' <- mapM (evalChanOffer we) chanoffs
-     expand chsets $ BNenable (BNbexpr we bexp1) chanoffs' (BNbexpr we bexp2)
+     case Data.Either.partitionEithers chanoffs' of
+        ([], r) -> expand chsets $ BNenable (BNbexpr we bexp1) r (BNbexpr we bexp2)
+        (s, _)  -> do IOB.putMsgs [ EnvData.TXS_CORE_MODEL_ERROR
+                                    ("Expand: Eval failed in expand - Enable" ++ show s) ]
+                      return []
 
   where
 
-     evalChanOffer :: WEnv VarId -> ChanOffer -> IOB.IOB ChanOffer
+     evalChanOffer :: WEnv VarId -> ChanOffer -> IOB.IOB (Either String ChanOffer)
 
      evalChanOffer _ (Quest vid) =
-          return $ Quest vid
+          return $ Right (Quest vid)
 
      evalChanOffer we' (Exclam vexp)  =  do
           tds <- gets IOB.tdefs
-          wal <- Eval.eval $ subst (Map.map cstrConst we') (funcDefs tds) vexp
-          return $ Exclam (cstrConst wal)
-
+          case TxsDefs.eval $ subst (Map.map cstrConst we') (funcDefs tds) vexp of
```

Comment:
Why not use `right` from `Control.Arrow`:

```haskell
let res = TxsDefs.eval $ subst (Map.map cstrConst we') (funcDefs tds) vexp of
-- right :: a b c -> a (Either d b) (Either d c) 
return $ right (Exclam . cstrConst) res
```

Reply:
signature is incomprehensible to me.
Please explain it to me next Monday face 2 face!